src/moodle_downloader.py

```diff
@@ -0,0 +1,84 @@
+import json
+import re
+import logging
+
+import globals
+import course_retrieval
+
+
+def list_resources(args):
+    try:
+        course_name = args.course
+        if course_name == "*":
+            # List the names of all available courses
+            course_retrieval.list_courses()
+        else:
+            # List all available resources within the specified course
+            course = course_retrieval.get_course(course_name)
+            if course is not None:
+                course.list_all_resources()
+    except:
```

Comment:
Have you determined what exactly throws an exception and what exception is thrown? Might be better to try explicitly handle it instead of wrapping the whole block in a `try` `except`

Reply:
This very coarse exception handling was a quick fix for [issue 11](https://github.com/NewLordVile/tum-moodle-downloader/issues/11). The exception that occurred can be found in the description of the issue. I am planning to implement more sophisticated error handling in the near future. I'll also include logging of the exceptions and will then refine the error handling based on the exception logs.